pkg/api/openshiftcluster.go

```diff
@@ -32,6 +35,25 @@ func (oc *OpenShiftCluster) UsesWorkloadIdentity() bool {
 	return oc.Properties.PlatformWorkloadIdentityProfile != nil && oc.Properties.ServicePrincipalProfile == nil
 }
 
+// ClusterMsiResourceId returns the resource ID of the cluster MSI or an error
+// if it encounters an issue while grabbing the resource ID from the cluster
+// doc. It is written under the assumption that there is only one cluster MSI
+// and will have to be refactored if we ever use more than one.
+func (oc *OpenShiftCluster) ClusterMsiResourceId() (*arm.ResourceID, error) {
+	if oc.Identity == nil || oc.Identity.UserAssignedIdentities == nil || len(oc.Identity.UserAssignedIdentities) == 0 {
```

Comment:
I considered different ways of going about this issue, and I decided on having the function return a boolean rather than a list of identities, since at least for now we expect the cluster doc to only have one identity. Let me know if that works from your perspective.